### PR TITLE
fix: Correctly calculate `Content-Length` when reading from a stream

### PR DESCRIPTION
**Pull request by arjankowski**

### CODE REVIEW

File: src/main/java/com/box/sdk/BinaryBodyUtils.java [hunk 1]
```diff
@@ -73,12 +74,36 @@ static void writeStreamWithContentLength(BoxAPIResponse response, OutputStream o
...
+     * Get the content length from the API response.
+     * In some cases, the Content-Length is not provided in the response headers.
+     * This could happen when getting the content representation for a compressed data.
+     * In that case the API will switch to chunk mode and provide the length in the "X-Original-Content-Length" header.
+     *
+     * @param response API response.
+     * @return Content length.
+     */
+    private static long getContentLengthFromAPIResponse(BoxAPIResponse response) {
+        long length = response.getContentLength();
```
Comment (abdulhaleem) [inline src/main/java/com/box/sdk/BinaryBodyUtils.java:93]: This checks if the body has length > 0

File: src/main/java/com/box/sdk/BinaryBodyUtils.java [hunk 2]
```diff
@@ -73,12 +74,36 @@ static void writeStreamWithContentLength(BoxAPIResponse response, OutputStream o
...
+        try {
+            if (length == -1 && response.getHeaders().containsKey(X_ORIGINAL_CONTENT_LENGTH)) {
```
Comment (abdulhaleem) [inline src/main/java/com/box/sdk/BinaryBodyUtils.java:95]: The code on line 93 checks for the size of the body, could you add one more checked please here 
for the header response.getHeaders().containsKey(HttpHeaders.CONTENT_LENGTH) as well please. Basically, 
1. Check for response.getContentLength()
2. Check for response.getHeaders().containsKey(HttpHeaders.CONTENT_LENGTH)
3. Check for response.getHeaders().containsKey(X_ORIGINAL_CONTENT_LENGTH)

Comment (arjankowski) [reply src/main/java/com/box/sdk/BinaryBodyUtils.java]: Done

**Review by abdulhaleem — APPROVED** [review]

File: src/main/java/com/box/sdk/BinaryBodyUtils.java [hunk 3]
```diff
@@ -9,6 +10,7 @@
  */
 final class BinaryBodyUtils {
     private static final int BUFFER_SIZE = 8192;
+    private static final String X_ORIGINAL_CONTENT_LENGTH = "X-Original-Content-Length";
```
Comment (lukaszsocha2) [inline src/main/java/com/box/sdk/BinaryBodyUtils.java:13]: pls add it to HttpHeaders class

**Review by lukaszsocha2 — APPROVED** [review]

**Pull request merged**